amundsen_application/static/js/utils/dateUtils.ts

```diff
@@ -0,0 +1,41 @@
+import * as Moment from 'moment-timezone';
+import AppConfig from 'config/config';
+
+const timezone = Moment.tz.guess();
+
+interface DateConfig {
```

Comment:
nit: it seems like what we'd need is 3 different types:
```
interface SomeName {
    timestamp: number
}

interface SomeNameEpoch {
    epochTimestamp: number
}

interface SomeNameDateString {
    dateString: string
    dateStringFormat: string
}
```
This is an idea specifically for developer friendliness because since all the values are optional, folks still have to come look at the `getMomentDate` logic to understand how to use it, and they won't get any helpful error messages when compiling if they use this incorrectly. That being said I think this is fine to merge because I'd want to look up the right way to create something like this on TypeScript. One caveat for this idea would be understanding the right way to know which option was passed in. https://stackoverflow.com/questions/14425568/interface-type-check-with-typescript